src/engine/strat_engine/blockdevmgr.rs

```diff
@@ -40,6 +40,40 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<HashSet<Device>> {
         .collect()
 }
 
+#[derive(Debug)]
+pub struct BlkDevSegment {
+    pub uuid: DevUuid,
+    pub segment: Segment,
+}
+
+impl BlkDevSegment {
+    pub fn new(uuid: DevUuid, segment: Segment) -> BlkDevSegment {
+        BlkDevSegment { uuid, segment }
+    }
+
+    pub fn to_segment(&self) -> Segment {
+        // TODO: dm-rs Segment should implement Clone & Copy.
```

Comment:
It should definitely implement ```Clone```, maybe not ```Copy```. I don't know where you should draw the line at ```Copy```, but I think it should be pretty small.

Reply:
We can address this if/when dm-rs makes this change.